pmd-plsql/etc/grammar/PLSQL.jjt

```diff
@@ -2721,23 +2721,16 @@ ASTMergeStatement MergeStatement() :
         |   [ LOOKAHEAD(2) SchemaName() "." ] TableName() [ TableAlias() ]
     )
     <ON> "(" Condition() ")"
-    [ LOOKAHEAD(MergeUpdateClausePrefix()) MergeUpdateClause() ]
-    [ LOOKAHEAD(2) MergeInsertClause() ]
+    ( <WHEN> ( ( KEYWORD("MATCHED") MergeUpdateClause() )  | ( <NOT> KEYWORD("MATCHED") MergeInsertClause() ) ) ) +
```

Comment:
The "+" at the end means, there must be either a merge update clause or a merge insert clause.... I think, it should be "*", because it's optional

```suggestion
// The order of the insert and update clause of a merge is not fixed
// (in contradiction to what the graphical schema's indicate in Oracle documentation)
( LOOKAHEAD(MergeUpdateClausePrefix()) MergeUpdateClause() | LOOKAHEAD(2) MergeInsertClause() )*
```

Reply:
The docs say;  
"You can specify the merge_insert_clause by itself or with the merge_update_clause. If you specify both, then they can be in either order."
To me that says that it's not optional and actually "insert" seems required. But as I'm not sure (certainly not for all possible Oracle versions), this seems to be the most robust (I only want to prevent a syntax error).  
I probably could make it something like: I[U] | UI
but maybe that would be more limiting that the actual Oracle parser.

The '+' symbol could be replaced with "{1,2}"  to limit the number of repetitions.  But that would still allow 2 inserts or 2 updates.

Reply:
I think we can be a bit relaxed here, if we are not sure how the actual Oracle parser behaves. Let's keep it optional and allow all possible variants (0 merge update/insert clause, merge update only, merge insert only, or both in either order).
That should allow us to parse without problems.
As PMD, we anyway expect that the code we parse is correct - we don't try to find compile errors or so.